### ISSUE TITLE
Incorrect inherent method resolution with multiple impls

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust-analyzer#9938]
```rust
struct S<T>(T);

trait Tr {}

impl<T: Tr> S<T> {
    fn call(&self, arg: u8) {}
}

impl S<()> {
    fn call(&self) {}
}

fn main() {
    S(()).call();
}
```

This gets a false-positive `mismatched-arg-count` diagnostic, because we choose the first `impl` block (despite the missing `(): Tr` impl), while the second one is actually correct.

This causes issues when using wasmer or wasm3, since they expose function handles with impls like these.

**Comment by lowr** [comment]
This is another case of #5441. Closing as it's been fixed :smile:
